sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ExchangeCoordinator.scala

```diff
@@ -232,7 +232,7 @@ class ExchangeCoordinator(
       // number of post-shuffle partitions.
       val partitionStartIndices =
         if (mapOutputStatistics.length == 0) {
-          None
+          Some(Array[Int]())
```

Comment:
I think we don't need to use `Option` here.

Reply:
Is it ok to ignore the `minNumPostShufflePartitions` value here?

Reply:
1. If we don't use Some(Array[Int]()) here, the number of post-shuffle partitions will be spark.sql.shuffle.partitions.
2. mapOutputStatistics.length == 0 indicates estimatePartitionStartIndices will not be invoked, so minNumPostShufflePartitions will be ignored anyway.
In my opition, all the number of pre-shuffle partitions is 0, the number of post-shuffle partitions should be 0 rather than spark.sql.adaptive.minNumPostShufflePartitions or spark.sql.shuffle.partitions.

Reply:
The `Option` is passed to `ShuffledRowRDD`. IIUC `None` indicates coordinator is not used and `spark.sql.shuffle.partitions` decides the number of reducers.

Reply:
I meant;
```
      val partitionStartIndices =
        if (mapOutputStatistics.length == 0) {
           Array.empty[Int]
        } else {
          estimatePartitionStartIndices(mapOutputStatistics)
        }

      ...
          exchange.preparePostShuffleRDD(shuffleDependencies(k), Some(partitionStartIndices))
```

Reply:
partitionStartIndices  should be an Option[Array[Int]] as the second parameter of ShuffleExchange.preparePostShuffleRDD. If use Array.empty[Int], a syntax error appears.